accounts/views.py

```diff
@@ -0,0 +1,141 @@
+                # if remember me is not checked on login
+                request.session.set_expiry(0)
+                request.session.modified = True
+            messages.success(request, 'Login Successful!')
+        else:
+            request.session['email'] = email
+            messages.error(
+                request, 'Incorect Email/Password! Please try again!')
+            return redirect("user_profile:index")
+        return redirect("user_profile:dashboard")
+
+
+class Logout(View, LoginRequiredMixin):
+    """ VIew for Handling logout
+    Only supports Post requests
+    logouts the user when endpoint is accessed else
+    throws error. User must be logged in to logout
+    """
+    login_url = 'user_profile:index'
+
+    def post(self, request, *args, **kwargs):
+        try:
+            logout(request)
+            messages.success(request, 'Logout Successful!')
+        except Exception as e:
+            messages.error(request, "Something went Wrong! {}".format(e))
+        return redirect('user_profile:index')
+
+
+class Register(View):
+    """View for User registration
+    only supports POST Method
+    creates the necessary db objects
+    logs the user in and redirect the user
+    to the dashboard else redirects back to
+    signup page
+    """
+
+    def post(self, request, *args, **kwargs):
+        registered_feed_content = '{full_name} has joined UBook'
+        if 'email' in request.session:
+            del request.session['email']
+
+        email = request.POST['email']
+        password1 = request.POST['password1']
+        password2 = request.POST['password2']
+        first_name = request.POST['firstname']
+        last_name = request.POST['lastname']
+        birthday = request.POST['birthday']
+        gender = request.POST['gender']
+        if password1 != password2:
+            messages.warning(
+                request, "Password and Confirm Password do not Match!")
+            return redirect('user_profile:index')
+        email = managers.CustomUserManager.normalize_email(email)
+        user = None
+        try:
+            user = models.CustomUser.objects.create(
+                email=email, first_name=first_name, last_name=last_name, is_active=True)
+            user.set_password(password1)
+            user_profile = UserProfile.objects.create(
+                user=user, birthday=birthday, gender=gender)
+            registered_feed_content = registered_feed_content.format(
+                full_name=user.get_full_name())
+            register_feed_template = FeedTemplate.objects.create(
+                content=registered_feed_content, feed_type='register')
+            register_feed_object = Feed.objects.create(
+                user=user, feed_template=register_feed_template)
+            user.save()
+            user_profile.save()
+            register_feed_template.save()
+            register_feed_object.save()
```

Comment:
We dont need to call .save() on an object created with .objects,create()

Reply:
fixed

Reply:
resolved